src/pypdfium2/_helpers/page.py

```diff
@@ -0,0 +1,430 @@
+            rotation (int):
+                A rotation value in degrees (0, 90, 180, or 270), in addition to page rotation.
+            
+            crop (typing.Tuple[float, float, float, float]):
+                Amount in PDF canvas units to cut off from page borders (left, bottom, right, top).
+                Crop is applied after rotation.
+            
+            colour (typing.Optional[ typing.Tuple[int, int, int, typing.Optional[int]] ]):
+                Page background colour. Defaults to white.
+                It can either be :data:`None`, or values of red, green, blue, and alpha ranging from 0 to 255.
+                If :data:`None`, the bitmap will not be filled with a colour, resulting in transparent background.
+                For RGB, 0 will include nothing of the colour in question, while 255 will completely include it. For Alpha, 0 means full transparency, while 255 means no transparency.
+            
+            annotations (bool):
+                Whether to render page annotations.
+            
+            greyscale (bool):
+                Whether to render in greyscale mode (no colours).
+            
+            optimise_mode (OptimiseMode):
+                How to optimise page rendering.
+        
+        Returns:
+            (BitmapDataHolder, str, (int, int)):
+            Bitmap data holder, colour format, and image size.
+            The colour format can be ``BGRA``, ``BGR``, or ``L``, depending on the parameters *colour* and *greyscale*.
+            Image size is given in pixels as a tuple of width and height.
+        """
+        
+        if colour is None:
+            fpdf_colour, use_alpha = None, True
+        else:
+            fpdf_colour, use_alpha = colour_tohex(*colour)
+        
+        cl_format, cl_pdfium = get_colourformat(use_alpha, greyscale)
+        n_colours = len(cl_format)
+        
+        form_config = pdfium.FPDF_FORMFILLINFO(2)
+        form_fill = pdfium.FPDFDOC_InitFormFillEnvironment(self._pdf.raw, form_config)
+        
+        src_width  = math.ceil(self.get_width()  * scale)
+        src_height = math.ceil(self.get_height() * scale)
+        if rotation in (90, 270):
+            src_width, src_height = src_height, src_width
+        
+        crop = [math.ceil(c*scale) for c in crop]
+        width  = src_width  - crop[0] - crop[2]
+        height = src_height - crop[1] - crop[3]
+        if any(d < 1 for d in (width, height)):
+            raise ValueError("Crop exceeds page dimensions (in px): width %s, height %s, crop %s" % (src_width, src_height, crop))
+        
+        bitmap = pdfium.FPDFBitmap_CreateEx(width, height, cl_pdfium, None, width*n_colours)
+        if fpdf_colour is not None:
+            pdfium.FPDFBitmap_FillRect(bitmap, 0, 0, width, height, fpdf_colour)
+        
+        render_flags = 0
+        if annotations:
+            render_flags |= pdfium.FPDF_ANNOT
+        if greyscale:
+            render_flags |= pdfium.FPDF_GRAYSCALE
```

Comment:
Add more flags, e. g. to disable anti-aliasing. See https://pdfium.googlesource.com/pdfium/+/15d326e3d3fbbf178fde4c22ec8860f08b54362c/public/fpdfview.h#745